File: back/src/main/java/br/com/feliva/back/models/Funcionario.java
package br.com.feliva.back.models;

import br.com.feliva.sharedClass.db.Model;
import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import jakarta.validation.Valid;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@Entity
@Table(name="funcionario")
public class Funcionario extends Model<Integer> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_funcionario")
    private Integer idFuncionario;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "id_pessoa")
    @Valid
    private Pessoa pessoa;

    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "id_endereco")
    @Valid
    private Endereco endereco;

//    @NotNull(message = "Informe um status.")
    @Enumerated(EnumType.STRING)
    @Column(length = 3)
    private Status status;

    @Column(name = "dt_contratacao")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd/MM/yyyy", locale = "pt_br")
    private LocalDate dtContratacao;

    private String telefone;
    private String celular;

    @Override
    public Integer getMMId() {
        return this.idFuncionario;
    }
}

File: front/src/app/model/Funcionario.ts
import {FormControl, FormGroup, Validators} from "@angular/forms";
import {Endereco} from "./Endereco";
import {Pessoa} from "./Pessoa";
import {Status} from "./Status";

export class Funcionario {
  idFuncionario?: number;
  pessoa?: Pessoa;
  endereco?: Endereco;
  dtContratacao ?:string;
  status?:Status


  static CreateFormGroup(entity?: Funcionario) {
    return new FormGroup({
      idFuncionario: new FormControl(entity?.idFuncionario),
      pessoa: Pessoa.CreateFormGroup(entity?.pessoa),
      endereco: Endereco.CreateFormGroup(entity?.endereco),
      dtContratacao: new FormControl(entity?.dtContratacao,[Validators.required]),

    });

  }
}